gammapy/image/maps.py

```diff
@@ -255,26 +255,118 @@ def coordinates_pix(self, mode='center'):
             y, x = y - 0.5, x - 0.5
         else:
             raise ValueError('Invalid mode to compute coordinates.')
-
         return x, y
 
-    def coordinates(self, origin=0, mode='center'):
+    def coordinates(self, mode='center'):
         """
         Sky coordinate images.
+
         Parameters
         ----------
-        origin : {0, 1}
-            Pixel coordinate origin.
         mode : {'center', 'edges'}
             Return coordinate values at the pixels edges or pixel centers.
+
         Returns
         -------
         coordinates : `~astropy.coordinates.SkyCoord`
             Position on the sky.
         """
         x, y = self.coordinates_pix(mode=mode)
-        coordinates = pixel_to_skycoord(x, y, self.wcs, origin)
+        coordinates = pixel_to_skycoord(x, y, self.wcs, self.wcs_origin)
         return coordinates
+    
+    def _get_boundaries(self, skymap_ref, skymap):
+        """
+        Get boundary coordinates of one sky map in the pixel coordinate system
+        of another reference sky map.
+        """
+        ymax, xmax = skymap.data.shape
+        ymax_ref, xmax_ref = skymap_ref.data.shape
+
+        # transform boundaries in world coordinates
+        bounds = skymap.wcs.wcs_pix2world([0, xmax], [0, ymax], skymap.wcs_origin)
+        
+        # transform to pixel coordinats in the reference skymap
+        bounds_ref = skymap_ref.wcs.wcs_world2pix(bounds[0], bounds[1], skymap_ref.wcs_origin)
+
+        # round to nearest integer and clip at the boundaries
+        xlo, xhi = np.rint(np.clip(bounds_ref[0], 0, xmax_ref))
+        ylo, yhi = np.rint(np.clip(bounds_ref[1], 0, ymax_ref))
+        return xlo, xhi, ylo, yhi
+
+    def paste(self, skymap, method='sum'):
+        """
+        Paste smaller skymap into sky map. 
+
+        WCS specifications of both skymaps must be aligned. If not call
+        `SkyMap.reproject()` on one of the maps first.
+
+        Parameters
+        ----------
+        skymap: `SkyMap`
+            Smaller sky map to paste.
+        method : {'sum', 'replace'}, optional
+            Sum or replace total values with cutout values.
+        """
+        xlo, xhi, ylo, yhi = self._get_boundaries(self, skymap)
+        xlo_c, xhi_c, ylo_c, yhi_c = self._get_boundaries(skymap, self)
+
+        if method == 'sum':
+            self.data[ylo:yhi, xlo:xhi] += skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
+        elif method == 'replace':
+            self.data[ylo:yhi, xlo:xhi] = skymap.data[ylo_c:yhi_c, xlo_c:xhi_c]
+        else:
+            raise ValueError('Invalid method: {0}'.format(method))
```

Comment:
Change `{0}` to `{}` (for consistency with all other string format calls in Gammapy).
(The position index is only needed for Python 2.6, which we don't support any more)